spec/system/contributors/conversations_spec.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Conversation interactions', js: false do
+  let(:user) { create(:user) }
+  let(:contributor) { create(:contributor, email: 'contributor@example.org') }
+  let(:first_received_message) do
+    create(:message, text: 'Message with the contributor as recipient 1', recipient: contributor)
+  end
+  let(:sent_message) do
+    create(:message, sender: contributor)
+  end
+  let(:last_received_message) do
+    create(:message, text: 'Message with the contributor as recipient 2', recipient: contributor)
+  end
+
+  before do
+    first_received_message
+    sent_message
+    last_received_message
+    visit conversations_contributor_path(contributor, as: user)
+  end
+
+  it 'can navigate to the requests view and back' do
+    expect(page).to have_text(first_received_message.text)
+    first_message_request_path = request_path(first_received_message.request, anchor: "contributor-#{contributor.id}")
+    puts first_received_message.request.title
+    find("a[href='#{first_message_request_path}']", text: first_received_message.request.title).trigger('click')
+
+    expect(page).to have_current_path(first_message_request_path.split('#').first)
```

Comment:
nope doesn't work here

Reply:
```suggestion
        expect(page).to have_current_path(request_path(first_received_message.request))
```

Ok, sorry I was reviewing the specs quickly as I was running late to leave the house. Now I can see what is happening. `first_message_request_path` is a local variable that includes the anchor and is used to find the anchor in the HTML. 

We actually do care that the path includes the path with the anchor as that is what should happen; however, for some reason the current_path does not include it. current_url does but then we run into an issue with the domain not matching. So, instead of spending time on it, I would suggest reducing the complexity and simply checking the current_path against the request_path.